src/Runner.Worker/ExecutionContext.cs

```diff
@@ -270,17 +277,37 @@ public void RegisterPostJobStep(IStep step)
         /// Helper function used in CompositeActionHandler::RunAsync to
         /// add a child node, aka a step, to the current job to the Root.JobSteps based on the location. 
         /// </summary>
-        public void RegisterNestedStep(IStep step, DictionaryContextData inputsData, int location, Dictionary<string, string> envData)
+        public IStep RegisterNestedStep(
+            IActionRunner step, DictionaryContextData inputsData, int location,
+            Dictionary<string, string> envData,
+            Boolean cleanUp = false)
         {
             // TODO: For UI purposes, look at figuring out how to condense steps in one node => maybe use the same previous GUID
             var newGuid = Guid.NewGuid();
-            step.ExecutionContext = Root.CreateChild(newGuid, step.DisplayName, newGuid.ToString("N"), null, null);
+
+            // Set Scope Name. Note, for our design, we consider each step in a composite action to have the same scope
+            // This makes it much simpler to handle their outputs at the end of the Composite Action
+            var childScopeName = !string.IsNullOrEmpty(this.ScopeName) ? $"{this.ScopeName}.{this.ContextName}" : this.ContextName;
```

Comment:
i think there's a bug here for the future (nested composite actions)

Should generate `__<GUID>` only when context name is empty.

Reply:
```
var safeContextName = !string.IsNullOrEmpty(ContextName) ? ContextName : $"__{newGuid}";
var childScopeName = !string.IsNullOrEmpty(this.ScopeName) ? $"{this.ScopeName}.{safeContextName}" : safeContextName;
```

is this ^ right? i'm getting confused now because childContextName is passed to Root.CreateChild and i dont know what that's for....


OK i just realized i'm way confused b/c i was getting confused whether in the for loop yet or not... i'll leave these notes here and will think through again later.